OpenHPL/Waterway/AirCushionSurgeTank.mo

```diff
@@ -0,0 +1,79 @@
+  parameter Modelica.SIunits.Height h_0 = 2 "Initial water height in the surge tank" annotation (
+    Dialog(group = "Initialization"));
+  input Modelica.SIunits.Pressure p_c0 = 4*para.p_a "Initial pressure of air-cushion inside the surge tank" annotation (
+    Dialog(group = "Geometry"));
+  //// possible parameters for temperature variation. Not finished...
+  //parameter Boolean TempUse = para.TempUse "If checked - the water temperature is not constant" annotation (Dialog(group = "Initialization"));
+  //parameter Modelica.SIunits.Temperature T_i = para.T_i "Initial water temperature in the pipe" annotation (Dialog(group = "Initialization", enable = TempUse));
+  //// variables
+  Modelica.SIunits.Mass m "water mass";
+  Modelica.SIunits.Momentum M "water momuntum";
+  Modelica.SIunits.Force M_dot "difference in influent and effulent momentum";
+  Modelica.SIunits.Force F "total force acting in the surge tank";
+  Modelica.SIunits.Area A = (pi*D ^ 2) / 4 "cross sectional area of the surge tank";
+  Modelica.SIunits.Length l = h / cos_theta "length of water in the surge tank";
+  Real cos_theta = H / L "slope ratio";
+  Modelica.SIunits.Velocity v "water velocity";
+  Modelica.SIunits.Force F_p "pressure force";
+  Modelica.SIunits.Force F_f "friction force";
+  Modelica.SIunits.Force F_g "friction force";
+  Modelica.SIunits.Pressure p_c "air-cushion force";
+  //// initial values for differential variables
+  Modelica.SIunits.Height h(start = h_0, fixed = true) "water height in the surge tank";
+  Modelica.SIunits.VolumeFlowRate V_dot(start = V_dot0, fixed = true) "water flow rate";
+  //// variables for temperature. Not in use for now...
+  //Real W_f, W_e;
+  //// conector (acquisition of algebraic variable, mass flow rate m_dot, and node pressure (manifold pressure) p_n)
+  extends OpenHPL.Interfaces.ContactNode;
+initial equation
+  if SteadyState == true then
+    der(M) = 0;
+    der(m) = 0;
+    //der(T_n) = 0;
+  else
+    h = h_0;
+    V_dot = V_dot0;
+    //T_n = T_i;
+  end if;
+equation
+  //// mass and momentum balance
+  der(m) = m_dot;
+  der(M) = M_dot+F;
+
+  m = para.rho * A * l;
+  m_dot = para.rho*V_dot;
+
+  M = m * v;
+  v = V_dot / A;
+  M_dot = m_dot*v;
+  F = F_p-F_f-F_g;
+  F_p = (p_n - p_c)*A;
+  F_f = Functions.DarcyFriction.Friction(v, D, l, para.rho, para.mu, eps);
+  F_g = m * para.g * cos_theta;
+ annotation (
+    Documentation(info="<html>
+<p>The simple model of the air cushion surge tank, which described by the momentum and mass differential equations. The mass balance depends on inlet and outlet mass flow rates. The momentum balance depends on inlet momentum and pressure dorp through the surge pipe together with gravity and friction forces. The main defined variable are <i>V_dot_s</i> and <i>h_s&nbsp;</i>(the flow rate and water level in the surge tank). The air in the surge tank is taken at inital pressure of <i>p_2</i></p>
+<p><img src=\"modelica://OpenHPL/Resources/Images/surgepic.png\"/></p>
```

Comment:
Could we start wit the SVGs instead of pngs right away?

Reply:
Absolutely, but i have problems with _arrow_  not showing in right direction in svg of inkscape. i mean the vertically downward arrows shows horizontally in the inkscape.

Reply:
Could you add the svg too. So I can have a look at it.

Reply:
ACST.svg is added in the Resource/Images that depict image for air cushion surge tank.

Reply:
So I checked and it looks fine in Inkscape:
![image](https://user-images.githubusercontent.com/9332/69441200-8f3b7180-0d4a-11ea-9da7-14993b28fd6e.png)

What do you mean does not look right?

Reply:
Btw some lines are not aligned ;-)

Reply:
It consists of a mesh grid color for water and air. The svg image is not rendered (for example by the browser) for the mesh grid. But the problem in arrowhead is not occurring.

Reply:
Also maybe the colours should be matching.

Reply:
Sorry, I had arrowhead problems with different figures. But .svg image consisting of colors in mesh grid when opened by the browser I cannot see the color. The image looks like this
![2019-11-22_172811](https://user-images.githubusercontent.com/24632158/69442931-b182be80-0d4d-11ea-86bf-c3565d729d9c.png)

Reply:
I've posted a solution video in the Teams channel